core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala

```diff
@@ -35,6 +37,43 @@ class MetadataRequestTest extends BaseRequestTest {
   }
 
   @Test
+  def testClusterIdWithAllTopics() {
+    val v2MetadataResponse = sendMetadataRequest(MetadataRequest.allTopics(), 2)
+    val v2ClusterId = v2MetadataResponse.clusterId()
+    assertNotNull("clusterId should not be null", v2ClusterId)
+
+    val v1MetadataResponse = sendMetadataRequest(MetadataRequest.allTopics(), 1)
+    val v1ClusterId = v1MetadataResponse.clusterId()
+    assertEquals(s"v1 clusterId should be ${MetadataResponse.NO_CLUSTER_ID}", v1ClusterId, MetadataResponse.NO_CLUSTER_ID)
+  }
+
+  @Test
+  def testClusterIdWithNoTopics() {
+    val v2MetadataResponse = sendMetadataRequest(new MetadataRequest(List[String]().asJava), 2)
+    val v2ClusterId = v2MetadataResponse.clusterId()
+    assertNotNull("clusterId should not be null", v2ClusterId)
+
+    val v1MetadataResponse = sendMetadataRequest(MetadataRequest.allTopics(), 1)
+    val v1ClusterId = v1MetadataResponse.clusterId()
+    assertEquals(s"v1 clusterId should be ${MetadataResponse.NO_CLUSTER_ID}", v1ClusterId, MetadataResponse.NO_CLUSTER_ID)
+  }
+
+  @Test
+  def testClusterIdIsValidUUID() {
+    val metadataResponse = sendMetadataRequest(MetadataRequest.allTopics(), 2)
+    val clusterId = metadataResponse.clusterId()
+    val decodedUUID = DatatypeConverter.parseBase64Binary(clusterId)
```

Comment:
Yes, this is part of the JDK[1]. 
The only other option in JDK7 is to use the internal `sun.misc.BASE64Encoder`/ `sun.misc.BASE64Decoder` classes. Otherwise, we will need to implement our own or add a third party dependency.

Note that once we upgrade to Java 8, we can use the new `Base64` class [2].

1. https://docs.oracle.com/javase/7/docs/api/javax/xml/bind/DatatypeConverter.html
2. http://docs.oracle.com/javase/8/docs/api/java/util/Base64.html